### PR TITLE
feat(services): Add postcards.

### PR DESCRIPTION
**Pull request by jalcine**
**What**: This adds the bits for authentication into the postcard creation form.

**Why**: Outside of being part of the codelab, this gives us access to create new postcards for dispatch.

### CODE REVIEW

**Comment by jalcine** [comment]
CC: @pallavi

File: test/services/postcard.spec.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var expect  = require('chai').expect;
+var Angular = require('angular');
+var Sinon   = require('sinon');
```
Comment (pallavi) [inline test/services/postcard.spec.js]: nit: add a newline here. we separate the packages we require from the local files we require

File: app/create/controller.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  $scope.payload = null;
+
+  $scope.create = function (payload) {
+    $scope.error = null;
+    $scope.loading = true;
+    $scope.postcard = null;
+
+    Postcard.create(payload)
+    .then(function (postcard) {
+      $scope.payload = postcard;
```
Comment (pallavi) [inline app/create/controller.js]: typo, should be `$scope.postcard`

File: app/create/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+var Angular    = require('angular');
```
Comment (pallavi) [inline app/create/index.js]: nit: new line

File: app/create/index.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+var State      = require('./state');
+var Controller = require('./controller');
```
Comment (pallavi) [inline app/create/index.js]: nit: alphabetize vars

**Review by pallavi — APPROVED** [review]
LGTM! small style things and a typo

File: app/create/index.js (hunk 4 above)
Comment (jalcine) [reply app/create/index.js]: I'm not following here; is it _because_ that `State` and `Controller` are capitalized that it's a nit? 👁

File: test/services/postcard.spec.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+var Config  = require('../../config');
+
+require('angular-mocks');
+
+describe('postcard service', function () {
```
Comment (jesscxu) [inline test/services/postcard.spec.js]: nit: use arrow functions here and for all the ones below

File: test/services/postcard.spec.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var expect  = require('chai').expect;
+var Angular = require('angular');
+var Sinon   = require('sinon');
+
+var Config  = require('../../config');
```
Comment (jesscxu) [inline test/services/postcard.spec.js]: nit: in addition to pallavi's comments above, this should be aligned with `=` in its own group (all local files required)

File: test/services/postcard.spec.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var expect  = require('chai').expect;
```
Comment (jesscxu) [inline test/services/postcard.spec.js]: nit: use `const` instead of `var`

File: test/services/postcard.spec.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+  beforeEach(Angular.mock.inject(function ($injector) {
+    $q         = $injector.get('$q');
+    $rootScope = $injector.get('$rootScope');
+    API        = $injector.get('API');
+    Postcard   = $injector.get('Postcard');
+  }));
+
+  describe('create', function () {
+    it('calls the correct endpoint with the correct params', function () {
```
Comment (jesscxu) [inline test/services/postcard.spec.js]: nit: we usually add a new line between `describe` blocks and `it` blocks

File: app/create/controller.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  $scope.error = null;
+  $scope.payload = null;
+
+  $scope.create = function (payload) {
+    $scope.error = null;
+    $scope.loading = true;
+    $scope.postcard = null;
+
+    Postcard.create(payload)
+    .then(function (postcard) {
```
Comment (jesscxu) [inline app/create/controller.js]: nit: arrow functions

**Review by jesscxu — APPROVED** [review]
LGTM pending nit fixes I mentioned that should be applied to most of the files

**Comment by jesscxu** [comment]
@jalcine can you also make the PR title match the first commit message?

**Pull request merged**